optuna/samplers/_grid.py

```diff
@@ -108,18 +111,33 @@ def sample_relative(self, study, trial, search_space):
         # object is hard to get at the beginning of trial, while we need the access to the object
         # to validate the sampled value.
 
-        unvisited_grids = self._get_unvisited_grid_ids(study)
+        target_grids = self._get_unvisited_grid_ids(study)
 
-        if len(unvisited_grids) == 0:
-            raise ValueError(
-                "All grids have been evaluated. If you want to avoid this error, "
-                "please make sure that unnecessary trials do not run during "
-                "optimization by properly setting `n_trials` in `study.optimize`."
+        if len(target_grids) == 0:
+            # This case may occur with distributed optimization or trial queue. If there is no
```

Comment:
This is more of a note regarding the "trial queue", but this sampler might not work that well with it. That is, the `grid_id` system attribute is set on trials even when they are popped from the trial queue and will be interpreted by the sampler as if they consumed an entry from the grid. It might be a problem caused by the fact that `sample_relative` is called from queued trials. It's probably beyond the scope of this PR though.